### PR TITLE
fix: flaky test `onboarding @no-mmi doesn't make any network requests to infura before onboarding is completed/test-failure-screenshot.png` and `onboarding @no-mmi Clicks create a new wallet, accepts a secure password, reveals the Secret Recovery Phrase, confirm SRP/test-failure-screenshot.png`

### LINKED ISSUE #24602: Flaky Test: Multiple onboarding tests failing due to an error where the "Create a new wallet" button cannot be found.
### What is this about?

**Reason of flakiness:**
We are trying to find button on wrong screen! Need to check why we are not on expected screen.

Related tests:
- [ ] MetaMask onboarding @no-mmi Clicks create a new wallet, accepts a secure password, reveals the Secret Recovery Phrase, confirm SRP --> this one is a different error
- [X] Wallet Created Events @no-mmi are not sent when onboarding user who chooses to opt out metrics
- [ ] MetaMask onboarding @no-mmi doesn't make any network requests to infura before onboarding is completed --> this one is a different error
- [X] MetaMask onboarding @no-mmi Verify that the user has been redirected to the correct page after creating a password for their new wallet
- [X] MetaMask onboarding @no-mmi User enters the wrong password during password creation

Failure:
https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/81133/workflows/0b962d69-e9d4-4c3e-9ca0-cfa829b667e2/jobs/2874094/tests

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/81118/workflows/04fa31d6-ea90-4e1d-9161-0e61fb05b85e/jobs/2873306/tests

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/81133/workflows/0b962d69-e9d4-4c3e-9ca0-cfa829b667e2/jobs/2874102/tests

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/81161/workflows/6a426c81-7fda-427d-b5de-2088b70ef326/jobs/2875750/tests

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/81169/workflows/dd9b42f9-1082-4dfa-a40d-4912b6deacbf/jobs/2876149/tests

Error message
Screenshot 2024-05-20 at 10 55 43

Screenshot 2024-05-20 at 11 36 43

Screenshot 2024-05-20 at 11 43 55

Screenshot 2024-05-20 at 13 51 53

Screenshot
Screenshot 2024-05-20 at 10 55 52

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by seaona**
## **Description**
This PR fixes more onboarding flakiness `onboarding @no-mmi doesn't make any network requests to infura before onboarding is completed/test-failure-screenshot.pn` and `onboarding @no-mmi Clicks create a new wallet, accepts a secure password, reveals the Secret Recovery Phrase, confirm SRP/test-failure-screenshot.png` .

The error is originated when  we try to click an element which is in stale state `StaleElementReferenceError: stale element reference: stale element not found in the current frame`.
This is a race condition originated in the `clickElement` method, in the following way:
1. we try to find the clickable element
2. the element is refreshed in between
3. right after, we try to click the element from 1 --> this results in the element being stale, as the element we are trying to click is an old instance (1) instead of (2)

Screenshot from 2024-05-28 14-43-54

Extra note: this race condition is surfaced in the Onboarding tests mostly (I've only seen it fail there, so far), since I suspect that the the actions we perform there, all refresh the same react component for onboarding, giving a window for this casuistic to happen

- Circle ci job failures: https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/83037/workflows/b435d63d-e50c-4dcd-af7d-4f6e0856ab6c/jobs/2967823/artifacts
- Circle ci logs:

Screenshot from 2024-05-28 14-37-47

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24813?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24602 (remaining items)

## **Manual testing steps**

1. Check ci or run onboarding tests multiple times (though I've never been able to repro this locally)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by pedronfigueiredo — APPROVED** [review]

**Review by chloeYue — APPROVED** [review]

**Review by DDDDDanica — APPROVED** [review]

File: test/e2e/webdriver/driver.js [hunk 1]
```diff
@@ -390,9 +390,23 @@ class Driver {
...
+      try {
+        const element = await this.findClickableElement(rawLocator);
+        await element.click();
+        return;
+      } catch (error) {
+        if (
+          error.name === 'StaleElementReferenceError' &&
+          attempt < retries - 1
+        ) {
+          await this.driver.delay(1000);
```
Comment (Gudahtt) [inline test/e2e/webdriver/driver.js:404]: Was this what was intended?
```
await this.delay(1000);
```

I'm not seeing a `delay` function on the internal webdriver instance. I'm seeing a `this.driver.delay is not a function` error on various e2e jobs

Comment (seaona) [reply test/e2e/webdriver/driver.js:404]: oops that is a great catch! indeed it should be `this.delay()`

Comment (seaona) [reply test/e2e/webdriver/driver.js:404]: here is the fix
https://github.com/MetaMask/metamask-extension/pull/24838

thank you v much Mark :pray:

Comment (seaona) [reply test/e2e/webdriver/driver.js:404]: on the bright side, this means that this race-condition was happening quite often 😅

**Pull request merged**